src/braket/experimental/autoqasm/program/program.py

```diff
@@ -12,17 +12,23 @@
 # language governing permissions and limitations under the License.
 
 """AutoQASM Program class, context managers, and related functions."""
+from __future__ import annotations
 
 import contextlib
 import threading
 from dataclasses import dataclass
 from enum import Enum
-from typing import Any, List, Optional, Union
+from typing import TYPE_CHECKING, Any, List, Optional, Tuple, Union
 
 import oqpy.base
 
 from braket.circuits.serialization import IRType
 from braket.experimental.autoqasm import constants, errors
+from braket.experimental.autoqasm.instructions.qubits import _qubit
+
+if TYPE_CHECKING:
+    from braket.experimental.autoqasm import Qubit
```

Comment:
Instead of importing `Qubit`, can you do this? Would this solve the reason you needed to add the `if TYPE_CHECKING:`?
```
from braket.experimental.autoqasm.instructions import QubitIdentifierType as Qubit
```